runtime/autoload/help.vim

```diff
@@ -0,0 +1,38 @@
+function! s:synnames(...) abort
+  if a:0
+    let [line, col] = [a:1, a:2]
+  else
+    let [line, col] = [line('.'), col('.')]
+  endif
+  return reverse(map(synstack(line, col), 'synIDattr(v:val,"name")'))
+endfunction
+
+function! HelpTopic()
+  let col = col('.') - 1
+  while col && getline('.')[col] =~# '\k'
+    let col -= 1
+  endwhile
+  let pre = col == 0 ? '' : getline('.')[0 : col]
+  let col = col('.') - 1
+  while col && getline('.')[col] =~# '\k'
+    let col += 1
+  endwhile
+  let post = getline('.')[col : -1]
+  let syn = get(synnames(), 0, '')
+  let cword = expand('<cword>')
+  if syn ==# 'vimFuncName'
+    return cword.'()'
+  elseif syn ==# 'vimOption'
+    return "'".cword."'"
+  elseif syn ==# 'vimUserAttrbKey'
+    return ':command-'.cword
+  elseif pre =~# '^\s*:\=$'
```

Comment:
Specifically for vim files open help with `:` if word happens to contain the first non-whitespace characters on the current line, or is preceded with `:` (there is a common convention to write `autocmd BufWritePre * :command`, same for some other complex commands like `:command` itself). Specifically for help files leave *only* the condition that current word is preceded with a colon.